// @swagger-api/apidom-reference/cjs/resolve/resolvers/HttpResolver.cjs
"use strict";

var _interopRequireWildcard = require("@babel/runtime-corejs3/helpers/interopRequireWildcard").default;
var _interopRequireDefault = require("@babel/runtime-corejs3/helpers/interopRequireDefault").default;
exports.__esModule = true;
exports.default = void 0;
var _stampit = _interopRequireDefault(require("stampit"));
var _Resolver = _interopRequireDefault(require("./Resolver.cjs"));
var url = _interopRequireWildcard(require("../../util/url.cjs"));
var _NotImplementedError = _interopRequireDefault(require("../../util/errors/NotImplementedError.cjs"));
const HttpResolver = (0, _stampit.default)(_Resolver.default, {
  props: {
    timeout: 5000,
    redirects: 5,
    withCredentials: false
  },
  init({
    timeout = this.timeout,
    redirects = this.redirects,
    withCredentials = this.withCredentials
  } = {}) {
    this.timeout = timeout;
    this.redirects = redirects;
    this.withCredentials = withCredentials;
  },
  methods: {
    canRead(file) {
      return url.isHttpUrl(file.uri);
    },
    async read() {
      throw new _NotImplementedError.default();
    },
    getHttpClient() {
      throw new _NotImplementedError.default();
    }
  }
});
var _default = HttpResolver;
exports.default = _default;
