// react-text-typing/dist/TextTyping.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _excluded = ["text", "showBlink", "speed", "component"];

var _templateObject, _templateObject2, _templateObject3;

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var blinking = (0, _styledComponents.keyframes)(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2.default)(["\n0% {\n  border-color: transparent;\n};\n  \n50% {\n  border-color: black;\n}\n"])));
var loading = (0, _styledComponents.keyframes)(_templateObject2 || (_templateObject2 = (0, _taggedTemplateLiteral2.default)(["\n0% {\n  max-width: 0%;\n}\n"])));

var TextRoot = _styledComponents.default.span(_templateObject3 || (_templateObject3 = (0, _taggedTemplateLiteral2.default)(["\nposition: relative;\ncolor: ", ";\nfont-size: 5em;\nmargin: 0;\npadding: 0;\nborder-right: .1em solid;\nanimation: ", " 0.8s linear infinite;\n\n&::before {\n  content: '", "';\n  position: absolute;\n  overflow: hidden;\n  white-space: nowrap;\n  animation: ", " ", "s steps(80);\n  max-width: 100%;\n  color: ", ";\n}\n"])), function (props) {
  return props.colorText;
}, blinking, function (props) {
  return props.internalText;
}, loading, function (props) {
  return props.timeTyping;
}, function (props) {
  return props.colorTyping;
});

var TextTyping = function TextTyping(_ref) {
  var text = _ref.text,
      showBlink = _ref.showBlink,
      speed = _ref.speed,
      Component = _ref.component,
      props = (0, _objectWithoutProperties2.default)(_ref, _excluded);

  var _useState = (0, _react.useState)(''),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      internalText = _useState2[0],
      setInternalText = _useState2[1];

  var timer = (0, _react.useRef)(null);
  (0, _react.useEffect)(function () {
    var typing = text.split('');
    timer.current = setInterval(function () {
      if (typing.length > 0) {
        var next = typing.shift();
        setInternalText(function (value) {
          return value + next;
        });
      } else {
        clearInterval(timer.current);
      }
    }, speed);
    return function () {};
  }, [text]);
  return /*#__PURE__*/_react.default.createElement(TextRoot, (0, _extends2.default)({
    as: Component,
    "data-text": text,
    internalText: internalText
  }, props), internalText);
};

TextTyping.propTypes = {
  /** Text to be demonstrated in the component */
  text: _propTypes.default.string.isRequired,

  /** Component to be used internally in the component */
  component: _propTypes.default.elementType,

  /** Text color */
  colorText: _propTypes.default.string,

  /** Background fill color */
  colorTyping: _propTypes.default.string,

  /** Show flashing text icon */
  showBlink: _propTypes.default.bool,

  /** Text speed appearing */
  speed: _propTypes.default.number,

  /** Time typing in animation */
  timeTyping: _propTypes.default.number
};
TextTyping.defaultProps = {
  component: 'h1',
  showBlink: true,
  colorText: '#FFF',
  colorTyping: '#0075D7',
  speed: 500,
  timeTyping: 10
};
var _default = TextTyping;
exports.default = _default;
